Unable to call 'show' function with 1D SharedArray

```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.7.0-DEV.5036 (2018-05-07 18:03 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit 9de318afa1* (1 day old master)
|__/                   |  x86_64-w64-mingw32

julia> versioninfo()
Julia Version 0.7.0-DEV.5036
Commit 9de318afa1* (2018-05-07 18:03 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i5 CPU         750  @ 2.67GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, nehalem)
Environment:

julia> using SharedArrays

julia> S = SharedArray{Int,1}([1,2,3]);

julia> show([1,2,3])
[1, 2, 3]
julia> show(S)
ERROR: AssertionError: ndims(X) != 1
Stacktrace:
 [1] show at .\arrayshow.jl:418 [inlined]
 [2] show(::Base.TTY, ::SharedArray{Int64,1}) at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v0.7\SharedArrays\src\SharedArrays.jl:481
 [3] show(::SharedArray{Int64,1}) at .\show.jl:314
 [4] top-level scope

julia>
```

I think the reason is that overloaded `show` function in `SharedArrays` is not exported properly. I found only `function show(io::IO, X::AbstractArray)` in `base/arrayshow.jl` has `@assert ndims(X) != 1` part.

**Edit**: I thought I found the same problem with 2D SharedArray, but acutally not. So I reverted the title.
I found another funciton `show_vector` and `show(io::IO, X::AbstractVector) = show_vector(io, X)` in `base/arrayshow.jl`.
```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.7.0-DEV.5036 (2018-05-07 18:03 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit 9de318afa1* (1 day old master)
|__/                   |  x86_64-w64-mingw32

julia> using SharedArrays

julia> S = SharedArray([1,2,3]);

julia> show(stdout, S)
ERROR: AssertionError: ndims(X) != 1
Stacktrace:
 [1] show at .\arrayshow.jl:418 [inlined]
 [2] show(::Base.TTY, ::SharedArray{Int64,1}) at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v0.7\SharedArrays\src\SharedArrays.jl:481
 [3] show(::SharedArray{Int64,1}) at .\show.jl:314
 [4] top-level scope

julia> Base.show_vector(stdout, S)
[1, 2, 3]
julia>
```

Why it fails calling `show(io::IO, X::AbstractVector)` even if `SharedArray{Int,1} <: AbstractVector` is `true` ?

I found another funciton `show_vector` and `show(io::IO, X::AbstractVector) = show_vector(io, X)` in `base/arrayshow.jl`, and it works with 1D `SharedArray`.
```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.7.0-DEV.5036 (2018-05-07 18:03 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit 9de318afa1* (1 day old master)
|__/                   |  x86_64-w64-mingw32

julia> using SharedArrays

julia> S = SharedArray([1,2,3]);

julia> show(S)
ERROR: AssertionError: ndims(X) != 1
Stacktrace:
 [1] show at .\arrayshow.jl:418 [inlined]
 [2] show(::Base.TTY, ::SharedArray{Int64,1}) at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v0.7\SharedArrays\src\SharedArrays.jl:481
 [3] show(::SharedArray{Int64,1}) at .\show.jl:314
 [4] top-level scope

julia> Base.show_vector(STDOUT, S)
WARNING: Base.STDOUT is deprecated, use stdout instead.
 in module Main
[1, 2, 3]
julia>
```

Why it fails calling `show_vector` with 1D `SharedArray` even if `SharedArray <: AbstractArray` is `true` ?

--- a/base/arrayshow.jl
+++ b/base/arrayshow.jl
@@ -415,7 +415,7 @@ _show_empty(io, X) = nothing # by default, we don't know this constructor
 
 # typeinfo aware (necessarily)
 function show(io::IO, X::AbstractArray)
-    @assert ndims(X) != 1
+    ndims(X) == 1 && return show_vector(io, X)
     prefix = typeinfo_prefix(io, X)
     io = IOContext(io, :typeinfo => eltype(X), :compact => true)
     isempty(X) ?
@@ -442,8 +442,6 @@ function show_vector(io::IO, v, opn='[', cls=']')
         show_delim_array(io, v, opn, ",", cls, false)
     end
 end
-
-show(io::IO, X::AbstractVector) = show_vector(io, X)
 
 
 ## Logic for displaying type information

--- a/stdlib/SharedArrays/test/runtests.jl
+++ b/stdlib/SharedArrays/test/runtests.jl
@@ -299,3 +299,7 @@ end
 let s = convert(SharedArray, [1,2,3,4])
     @test pmap(i->length(s), 1:2) == [4,4]
 end
+
+let S = SharedArray([1,2,3])
+    @test sprint(show, S) == "[1, 2, 3]"
+end

